dbt/adapters/bigquery/connections.py

```diff
@@ -283,7 +288,26 @@ def exception_handler(self, sql):
             raise DbtRuntimeError(exc_message)
 
     def cancel_open(self) -> None:
-        pass
+        names = []
+        this_connection = self.get_if_exists()
+        with self.lock:
+            for thread_id, connection in self.thread_connections.items():
+                if connection is this_connection:
+                    continue
+
+                if connection.handle is not None and connection.state == ConnectionState.OPEN:
+                    client = connection.handle
+                    for job_id in self.jobs_by_thread.get(thread_id, []):
+
+                        def fn():
+                            return client.cancel_job(job_id)
+
+                        self._retry_and_handle(msg=f"Cancel job: {job_id}", conn=connection, fn=fn)
+
+                    self.close(connection)
+
+                if connection.name is not None:
+                    names.append(connection.name)
```

Comment:
Do we need to return `names` here? if we do we need to re-add it and change the type for function.